src/core/user-wallet.ts

```diff
@@ -102,22 +102,36 @@ export abstract class UserWallet {
   }
 
   async getTransactions(): Promise<Array<ITransaction>> {
-    const rawTransactions = await this.getRawTransactions()
+    const rawTransactionsPromise = this.getRawTransactions()
+    const lightningFundingUserPromise = User.findOne({ role: "funder" })
+    const [rawTransactions, lightningFundingUser] = await Promise.all([
+      rawTransactionsPromise,
+      lightningFundingUserPromise,
+    ])
 
     return rawTransactions.map((item) => {
       const amount = item.credit - item.debit
       const isCredit = amount > 0
-      const isValidCreditMemo = isCredit && amount >= MEMO_SHARING_SATS_THRESHOLD
-      const isDebit = !isCredit
 
       const memoUsername = item.username
         ? isCredit
           ? `from ${item.username}`
           : `to ${item.username}`
         : null
 
-      const memoSpamFilter = (memoString) =>
-        memoString ? (isDebit || isValidCreditMemo ? memoString : null) : null
+      const memoSpamFilter = (memoString) => {
+        const creditBelowThreshold = isCredit && amount < MEMO_SHARING_SATS_THRESHOLD
+        const funderPayer =
+          item.username && item.username == lightningFundingUser?.username
+
+        const isCreditWithValidMemo = !creditBelowThreshold || funderPayer
```

Comment:
I think maybe we can have a rule for boolean variables, For ex: must have `is` or `has` prefix so in that way are easier to read and identify the type